task4/iot/apps/device/lib/device/scanner.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Device.Scanner do
+  use GenServer
+
+  def start_link(options \\ []) do
+    GenServer.start_link(__MODULE__, :ok, [options])
+  end
+    
+  def init(:ok) do
+    :dnssd.start
+    :dnssd.browse("_http._tcp")
+    {:ok, []}
+  end
+
+  def handle_info(msg, state) do
+	  {:dnssd, reference, {_, change,{name, type , domain}}} = msg
+    case change do
+      :add ->
+        {:ok, {_, port, _}} = :dnssd.resolve_sync name, type , domain
+        
+        device = %Device{
+                  name: name,
+                  domain: domain,
+                  data: "port=#{port}"
+                }
+        Device.Registry.add({reference, device})
```

Comment:
This `reference` is not unique per registered device - [it's a reference of the `browse` operation started in `init/1`](https://github.com/benoitc/dnssd_erlang#browsing-for-services). We can't use it as the key in device registry. Instead, we can use e.g. the reported name of the device or something like `name <> "." <> domain`.

I stumbled upon the problem when trying the display the list in a browser - only one device was ever listed, no matter how many I tried to register from my host.